project/src/main/java/com/google/sps/servlets/EventCreationServlet.java

```diff
@@ -30,30 +25,32 @@ public void doGet(HttpServletRequest request, HttpServletResponse response) thro
     Optional<Event> eventOptional = SpannerTasks.getEventById(eventId);
 
     // If event DNE, sends 404 ERR to frontend
-    if (!eventOptional.isPresent()) {
-        response.sendError(HttpServletResponse.SC_NOT_FOUND);
+    if (eventOptional.isPresent()) {
+      Event event = eventOptional.get();
+      response.setContentType("text/html;");
+      response.getWriter().println(new Gson().toJson(event));
     } else {
-        Event event = eventOptional.get().toBuilder().setId(eventId).build();
-        response.setContentType("text/html;");
-        response.getWriter().println(new Gson().toJson(event));
-    }  
+      response.sendError(HttpServletResponse.SC_NOT_FOUND);
```

Comment:
Here, I think you can add an error message with the error code like: string.format("No events found with event ID %s", eventId);

Reply:
Done